synapse/storage/databases/main/lock.py

```diff
@@ -216,6 +219,7 @@ async def try_acquire_read_write_lock(
                 lock_name,
                 lock_key,
                 write,
+                db_autocommit=True,
```

Comment:
Worth a comment? Why is it unhelpful to run this function in a transaction?

I think:

- we're still using isolation level REPEATABLE READ
- but we're not in a transaction, so if it fails we won't bother to retry
- which means we'll properly wait for the lock renewal period before trying to lock again, thrashing the DB less?

Reply:
Broadly: there's not much point doing a transaction for a single query as it basically has the same semantics, and by doing auto commit we reduce the work as we don't have to start/end the transaction.

Reply:
Oh: and means the length of time we hold the lock on those rows is reduced to just the query execution time, which helps reduce the likelihood of clashes